engine/apps/api/views/labels.py

```diff
@@ -25,9 +25,9 @@
 
 class LabelsFeatureFlagViewSet(ViewSet):
     def initial(self, request, *args, **kwargs):
-        super().initial(request, *args, **kwargs)
         if not is_labels_feature_enabled(self.request.auth.organization):
             raise NotFound
+        super().initial(request, *args, **kwargs)
```

Comment:
The `initial` method does a bunch of different things like auth, throttling, etc. If the feature flag is disabled, we want to return 404 right away, so I moved the super method call down